src/test/java/MockTest.java

```diff
@@ -0,0 +1,435 @@
+                .andExpect(status().isOk())
+                .andExpect(content().string(containsString("value=\"" + plate.getPaid().toString() + "\"")));
+    }
+
+    @Test
+    public void shouldHavePlatePricedFor2HoursNonVIP() throws Exception {
+        priceCheck(Instant.now().minus(Duration.ofMinutes(121)), Instant.now(), false);
+    }
+
+    @Test
+    public void shouldHavePlatePricedFor2HoursVIP() throws Exception {
+        priceCheck(Instant.now().minus(Duration.ofMinutes(121)), Instant.now(), true);
+    }
+
+    @Test
+    public void shouldHavePlatePricedFor24HoursNonVIP() throws Exception {
+        priceCheck(Instant.now().minus(Duration.ofMinutes(1441)), Instant.now(), false);
+    }
+
+    @Test
+    public void shouldHavePlatePricedFor24HoursVIP() throws Exception {
+        priceCheck(Instant.now().minus(Duration.ofMinutes(1441)), Instant.now(), true);
+    }
+
+    private Plate getPlate(String plateNr, Instant start, Instant end, boolean isVip, BigDecimal paid) {
+        Plate plate = new Plate();
+        plate.setPlateNr(plateNr);
+        plate.setStart(start);
+        plate.setEnd(end);
+        plate.setRegion(Region.PLN);
+        plate.setVip(isVip);
+        plate.setPaid(paid);
+        return plate;
+    }
+
+    @Test
+    public void shouldServiceReturnIncomeFor2PlatesHourEach() throws Exception {
+        final String EXAMPLE_PLATE_NR = "12345678";
+        Plate plate = getPlate(EXAMPLE_PLATE_NR, Instant.now(), Instant.now().plus(Duration.ofMinutes(1)), false, new BigDecimal(1));
+        Plate plate2 = getPlate(EXAMPLE_PLATE_NR, Instant.now(), Instant.now().plus(Duration.ofMinutes(1)), false, new BigDecimal(1));
+
+        plateRepository.save(plate);
+        plateRepository.save(plate2);
+
+        DateContainer dateContainer = new DateContainer();
+        dateContainer.setDateTime(LocalDate.now());
+        assertEquals(2, pricingService.getIncomeForDay(dateContainer), 0.0);
+    }
+
+    @Test
+    public void shouldOwnerSeeIncomeFor2PlatesHourEach() throws Exception {
+        final String EXAMPLE_PLATE_NR = "12345678";
+        Plate plate = getPlate(EXAMPLE_PLATE_NR, Instant.now(), Instant.now().plus(Duration.ofMinutes(1)), false, new BigDecimal(1));
+        Plate plate2 = getPlate(EXAMPLE_PLATE_NR, Instant.now(), Instant.now().plus(Duration.ofMinutes(1)), false, new BigDecimal(1));
+
+        plateRepository.save(plate);
+        plateRepository.save(plate2);
+
+        DateContainer dateContainer = new DateContainer();
+        dateContainer.setDateTime(LocalDate.now());
+        mockMvc.perform(post("/getIncome")
+                .param("dateTime", dateContainer.getDateTime().toString())
+                .param("income", Double.toString(pricingService.getIncomeForDay(dateContainer)))
+                .sessionAttr("dateContainer", dateContainer))
+                .andExpect(status().isOk())
+                .andExpect(content().string(containsString("<span>2.0</span>")));
```

Comment:
Czemu w aserecji sprawdzasz czy kwota jest w spanie?

Reply:
Hmmm, w zasadzie to te htmle nie będą posiadać nigdzie indziej ciągów typu: "2.0", "0.0", "1 PLN", "3.20 EUR", etc. więc tych spanów można się pozbyć.